namer/namer.cc

```diff
@@ -467,6 +467,12 @@ class SymbolFinder {
     }
 
     ast::TreePtr postTransformSend(core::Context ctx, ast::TreePtr tree) {
+        // Only consider sends at (or near) the top level of a class def, but **not** inside any
+        // other definition (like a method).
+        if (ownerStack.back().kind() != DefinitionKind::Class) {
```

Comment:
There's actually a way to do this that makes namer faster: Make this pass a `ShallowMap`!

Reply:
(We might be able to do the same with `TreeSymbolizer`? Not sure.)

Reply:
(Unless we otherwise care about definitions within method definitions? I don't know their semantics in Ruby.)

Reply:
+1 to this, I like the idea of skipping method bodies unless we actually have to traverse them.